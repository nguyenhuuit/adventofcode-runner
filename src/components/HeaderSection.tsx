import React from 'react';

import { Box, Text } from 'ink';

interface HeaderSectionProps {
  year: string;
  day: string;
  part: string;
  language: string;
  userName?: string;
  star?: string;
}

const HeaderSection = ({ year, day, part, language, userName, star }: HeaderSectionProps) => {
  return (
    <Box>
      <Text>📆 Year:</Text>
      <Text italic bold color={'#FF8800'}>
        {' '}
        {year}{' '}
      </Text>
      <Text>Day:</Text>
      <Text italic bold color={'#FF8800'}>
        {' '}
        {day}{' '}
      </Text>
      <Text>Part:</Text>
      <Text italic bold color={'#FF8800'}>
        {' '}
        {part}{' '}
      </Text>
      <Text>| 📘 Language:</Text>
      <Text italic bold color={'#FF8800'}>
        {' '}
        {language}{' '}
      </Text>
      {userName && (
        <>
          <Text>| 👤 Username:</Text>
          <Text italic bold color={'#FF8800'}>
            {' '}
            {userName}{' '}
          </Text>
        </>
      )}
      {star && (
        <Text italic bold color={'#FF8800'}>
          {star}⭐️{' '}
        </Text>
      )}
    </Box>
  );
};

export default HeaderSection;
