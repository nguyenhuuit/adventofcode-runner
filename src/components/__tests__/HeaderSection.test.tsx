import { render } from '@testing-library/react';

import React from 'react';

import HeaderSection from '../HeaderSection';

interface MockBoxProps {
  children: React.ReactNode;
  [key: string]: unknown;
}

interface MockTextProps {
  children: React.ReactNode;
}

jest.mock('ink', () => ({
  Box: ({ children, ...props }: MockBoxProps) => (
    <div data-testid="box" {...props}>
      {children}
    </div>
  ),
  Text: ({ children }: MockTextProps) => <span data-testid="text">{children}</span>,
}));

describe('HeaderSection', () => {
  it('renders correctly with all props', () => {
    const { container } = render(
      <HeaderSection
        year="2023"
        day="1"
        part="1"
        language="typescript"
        userName="test-user"
        star="1"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders correctly without optional props', () => {
    const { container } = render(
      <HeaderSection year="2023" day="1" part="1" language="typescript" />
    );
    expect(container).toMatchSnapshot();
  });
});
