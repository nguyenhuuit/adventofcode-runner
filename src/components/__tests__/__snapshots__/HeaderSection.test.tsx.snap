// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSection renders correctly with all props 1`] = `
<div>
  <div
    data-testid="box"
  >
    <span
      data-testid="text"
    >
      📆 Year:
    </span>
    <span
      data-testid="text"
    >
       
      2023
       
    </span>
    <span
      data-testid="text"
    >
      Day:
    </span>
    <span
      data-testid="text"
    >
       
      1
       
    </span>
    <span
      data-testid="text"
    >
      Part:
    </span>
    <span
      data-testid="text"
    >
       
      1
       
    </span>
    <span
      data-testid="text"
    >
      | 📘 Language:
    </span>
    <span
      data-testid="text"
    >
       
      typescript
       
    </span>
    <span
      data-testid="text"
    >
      | 👤 Username:
    </span>
    <span
      data-testid="text"
    >
       
      test-user
       
    </span>
    <span
      data-testid="text"
    >
      1
      ⭐️
       
    </span>
  </div>
</div>
`;

exports[`HeaderSection renders correctly without optional props 1`] = `
<div>
  <div
    data-testid="box"
  >
    <span
      data-testid="text"
    >
      📆 Year:
    </span>
    <span
      data-testid="text"
    >
       
      2023
       
    </span>
    <span
      data-testid="text"
    >
      Day:
    </span>
    <span
      data-testid="text"
    >
       
      1
       
    </span>
    <span
      data-testid="text"
    >
      Part:
    </span>
    <span
      data-testid="text"
    >
       
      1
       
    </span>
    <span
      data-testid="text"
    >
      | 📘 Language:
    </span>
    <span
      data-testid="text"
    >
       
      typescript
       
    </span>
  </div>
</div>
`;
